links/links.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package links contains the dynamic links sdks. Specifically get link stats.
+package links
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	"firebase.google.com/go/internal"
+	"google.golang.org/api/option"
+	"google.golang.org/api/transport"
+)
+
+const (
+	linksEndPoint     = "https://firebasedynamiclinks.googleapis.com/v1"
+	linksStatsRequest = "%s/linkStats?durationDays=%d"
+)
+
+// LinkStats is returned from the GetLinkStats, contains an array of Event Stats
+type LinkStats struct {
+	EventStats []EventStats `json:"linkEventStats"`
+}
+
+// EventStats will contain the counts for the aggregations for the requested period
+type EventStats struct {
+	Platform Platform  `json:"platform"`
+	ET       EventType `json:"event"`
+	Count    int32     `json:"count,string"`
+}
+
+// Platform constant "enum" for the event
+type Platform int
```

Comment:
I wonder what we actually gain by making this an int. If this was just an alias for string, you could just use the same constants used in the server-side. That means you don't have to maintain 2 maps for each enum type.

Reply:
ok